crates/circuit/src/packed_instruction.rs

```diff
@@ -514,10 +514,93 @@ pub struct PackedInstruction {
     /// requires the GIL to even `get` (of course!), which makes implementing `Clone` hard for us.
     /// We can revisit once we're on PyO3 0.22+ and have been able to disable its `py-clone`
     /// feature.
-    pub py_op: OnceLock<Py<PyAny>>,
+    py_op: OnceLock<Py<PyAny>>,
 }
 
 impl PackedInstruction {
+    /// Creates a new instance of #[PackedInstruction]
+    pub fn new(
+        op: PackedOperation,
+        qubits: Interned<[Qubit]>,
+        clbits: Interned<[Clbit]>,
+        params: Option<SmallVec<[Param; 3]>>,
+        extra_attrs: ExtraInstructionAttributes,
+    ) -> Self {
+        Self {
+            op,
+            qubits,
+            clbits,
+            params: params.map(Box::new),
+            extra_attrs,
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceLock::new(),
+        }
+    }
+
+    // Getters
```

Comment:
I don't think we have a pattern for labeling sections, and it's probably easy for these to fall out of sync with the code. I'd recommend we drop them ^^